src/utils/paths.cpp

```diff
@@ -30,11 +30,20 @@ std::pair<bool, std::string> resolve_path::normalizer(const char *parent, const
     memset(ret+parent_len,0,child_len+1);
     memcpy(ret,parent,parent_len);
     int ptr = parent_len;
-    
+
     for(int j=0; j<child_len; ){
         if(tkn(child+j,"../")){
             //Go back to the past /
             if(ptr<=1)return {false, ""}; //Early failure, cannot track back.
+            if (j > 0) {
+                if (child[j-1] != '/') {
+                    uint8_t i = 0;
+                    for(;i<=sizeof "../";i++) {
```

Comment:
Ok, so this you might laugh at. ;) Something just didn't feel right about using "3" here.

In my own code, I might `#define DOUBLE_DOT_DIR_SEP "../"` or something at the top, then just use that string... and then using `sizeof` would seem more appropriate, but that too seems like overkill.

Reply:
No, that is totally fine, hardcoded numbers are just bad. That is resolved at compile time so it is not like it reduces performance or anything.

For context, in modern C++ one would define a constexpr value in place of a define if it had to be made more explicit.

Reply:
Awesome! Maybe @Jammyjamjamman is right and I'm not a bad developer! :)